images/virtualization-artifact/pkg/controller/service/disk_service.go

```diff
@@ -116,7 +116,28 @@ func (s DiskService) Start(
 		return nil
 	}
 
-	return supplements.EnsureForDataVolume(ctx, s.client, sup, dvBuilder.GetResource(), s.dvcrSettings)
+	dv := dvBuilder.GetResource()
```

Comment:
The service should not know who it is interacting with. It doesn't know if it's being used by vi/vd or someone else. The management of vi/vd belongs to the business logic of the vi/vd handlers. This responsibility should be moved there.